src/GovUk.Education.ExploreEducationStatistics.Public.Data.Api.Tests/Controllers/PublicationsControllerTests.cs

```diff
@@ -0,0 +1,207 @@
+        int page,
+        int pageSize,
+        int numberOfPublishedDataSets,
+        int numberOfUnpublishedDataSets)
+    {
+        var client = SetupApp(new ContentApiClientMock()).CreateClient();
+
+        var publishedDataSets = GeneratePublishedDataSets(numberOfPublishedDataSets);
+        var unpublishedDataSets = GenerateUnublishedDataSets(numberOfUnpublishedDataSets);
+
+        var allDataSets = publishedDataSets.Concat(unpublishedDataSets);
+
+        await AddTestDataSets(allDataSets.ToArray());
+
+        var response = await ListPublications(client, page, pageSize);
+
+        Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+
+        var content = await response.Content.ReadFromJsonAsync<PaginatedPublicationListViewModel>();
+
+        Assert.NotNull(content);
+        Assert.Equal(page, content!.Paging.Page);
+        Assert.Equal(pageSize, content!.Paging.PageSize);
+        Assert.Equal(numberOfPublishedDataSets, content!.Paging.TotalResults);
+
+        var publishedDataSetPublicationIdsToBeReturned = allDataSets
+            .Where(ds => ds.Status == DataSetStatus.Published)
+            .Select(ds => ds.PublicationId)
+            .Skip((page - 1) * pageSize)
+            .Take(pageSize);
+
+        Assert.Equal(publishedDataSetPublicationIdsToBeReturned.Count(), content.Results.Count);
+
+        var publicationIdsThatShouldNotHaveBeenReturned = allDataSets
+            .Select(ds => ds.PublicationId)
+            .Except(publishedDataSetPublicationIdsToBeReturned);
+
+        foreach (var publicationId in publishedDataSetPublicationIdsToBeReturned)
+        {
+            Assert.Contains(content.Results, r => r.Id == publicationId);
+        }
+
+        foreach (var publicationId in publicationIdsThatShouldNotHaveBeenReturned)
+        {
+            Assert.DoesNotContain(content.Results, r => r.Id == publicationId);
+        }
+    }
+
+    [Fact]
+    public async Task ListPublications_NoPublishedDataSets_Returns200WithEmptyList()
+    {
+        var client = SetupApp(new ContentApiClientMock()).CreateClient();
+
+        var response = await ListPublications(client, 1, 1);
+
+        Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+
+        var content = await response.Content.ReadFromJsonAsync<PaginatedPublicationListViewModel>();
+
+        Assert.NotNull(content);
+        Assert.Equal(1, content!.Paging.Page);
+        Assert.Equal(1, content!.Paging.PageSize);
+        Assert.Equal(0, content!.Paging.TotalResults);
+        Assert.Empty(content!.Results);
+    }
+
+    [Theory]
+    [InlineData(0)]
+    [InlineData(-1)]
+    public async Task ListPublications_RequestedPageIsTooSmall_Returns400(int page)
+    {
+        var client = SetupApp(new ContentApiClientMock()).CreateClient();
+
+        var response = await ListPublications(client, page, 1);
+
+        Assert.Equal(HttpStatusCode.BadRequest, response.StatusCode);
```

Comment:
We could probably do a bit more to test the error response more, but my intention was to customise the default .NET error response a bit more to provide extra context for users.

Will discuss this with you further, but will probably try and handle this as a separate PR and make the appropriate changes to these tests as well.

Reply:
leaving this until later